pennylane/transforms/metric_tensor.py

```diff
@@ -400,27 +425,38 @@ def _metric_tensor_cov_matrix(tape, diag_approx):
     obs_list = []
     coeffs_list = []
     params_list = []
+    in_argnum_list = []
 
     for queue, curr_ops, param_idx, _ in graph.iterate_parametrized_layers():
         params_list.append(param_idx)
+        in_argnum_list.append([p in argnum for p in param_idx])
+
+        if not any(in_argnum_list[-1]):
+            # no tape needs to be created for this block
+            continue

```

Comment:
Yes, this is a good idea.

Reply:
Ah, now I remember why I did it the way it is now. The `metric_tensor_tapes` would need some dummy tape so that the `probs` argument in the `processing_fn`. I don't know whether there is a reasonable way to achieve this, importantly without requesting unnecessary quantum resources. Do you have any suggestions here?

Reply:
Okay, we definitely should not be creating dummy tapes, that is correct! Even if they were to be empty, they might end up clogging a queue or something like that.
This means that one index like `probs_idx` is required to access the correct result of the executed tapes.

I'm wondering whether we still might get away with removing some of the counters in the `_metric_tensor_hadamard` function. Maybe it's me, but tracking multiple counters makes it quite hard for me to read the function. What do you think? :)